llvm/lib/SYCLLowerIR/ESIMD/LowerESIMDSlmReservation.cpp

```diff
@@ -362,9 +364,14 @@ class ScopedCallGraph {
           if (auto *CB = dyn_cast<CallBase>(&I)) {
             if (isSlmInitCall(dyn_cast<CallInst>(CB))) {
               auto *CI = dyn_cast<CallInst>(CB);
-              esimd::assert_and_diag(!SlmInitCall,
-                                     "multiple slm_init calls in function ",
-                                     F.getName());
+              if (SlmInitCall) {
```

Comment:
Did the patch change anything except the error message? I mean, didn't the previous code with `esimd::assert_and_diag(!SlmInitCall` do the same job - report an error in exactly same conditions?

Reply:
The good check of slm_init calls should include:
1) slm_init is called from kernel, or from a function that is guaranteed to be inlined into the kernel
2) there is only 1 slm_init call in the kernel and all functions called from it.